application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,265 @@
+
+        if (ephemeral) {
+            event.replyEmbeds(makeEmbed(title, description))
+                .addActionRow(menu.build())
+                .setEphemeral(true)
+                .queue();
+        } else {
+            event.getChannel()
+                .sendMessageEmbeds(makeEmbed(title, description))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+        }
+    }
+
+    /**
+     * Adds role options to a selection menu
+     * <p>
+     *
+     * @param event the {@link SlashCommandEvent}
+     * @param menu the menu to add options to {@link SelectionMenu.Builder}
+     * @param placeHolder the placeholder for the menu {@link String}
+     * @param minValues the minimum number of selections. nullable {@link Integer}
+     */
+    private static void addMenuOptions(@NotNull SlashCommandEvent event,
+            @NotNull SelectionMenu.Builder menu, @NotNull String placeHolder,
+            @Nullable Integer minValues) {
+
+        Role highestBotRole =
+                Objects.requireNonNull(event.getGuild()).getSelfMember().getRoles().get(1);
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+
+        List<Role> roles = new ArrayList<>(
+                guildRoles.subList(guildRoles.indexOf(highestBotRole) + 1, guildRoles.size()));
+
+        if (minValues != null) {
+            menu.setMinValues(minValues);
+        }
+
+        menu.setPlaceholder(placeHolder).setMaxValues(roles.size());
+
+        for (Role role : roles) {
+            if (role.getName().equals("@everyone"))
+                continue;
+            menu.addOption(role.getName(), role.getId());
+        }
+    }
+
+    /**
+     * Creates an embedded message to send with the selection menu
+     *
+     * <p>
+     * </p>
+     *
+     * @param title for the embedded message. nullable {@link String}
+     * @param description for the embedded message. nullable {@link String}
+     * @return the formatted embed {@link MessageEmbed}
+     */
+    private static @NotNull MessageEmbed makeEmbed(@Nullable String title,
+            @Nullable String description) {
+
+        if (title == null) {
+            title = "Select your roles:";
+        }
+
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+        embedBuilder.setTitle(title);
+
+        if (description != null) {
+            embedBuilder.setDescription(description);
+        }
+
+        embedBuilder.setColor(new Color(24, 221, 136));
+
+        return embedBuilder.build();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+
+        // Get the roles the bot can interact with
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
```

Comment:
```suggestion
        List<Role> guildRoles = event.getGuild().getRoles();
```

All Objects#requireNonNull does is throw a NPE with less info.
So this is kinda useless since it'll throw a NPE either way.